lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/auth/DBOAuthenticationDAOImpl.java

```diff
@@ -193,6 +195,27 @@ public void setTermsOfUseAcceptance(long principalId, Boolean acceptance) {
 		basicDAO.createOrUpdate(agreement);
 	}
 	
+	@Override
+	@WriteTransaction
+	public void setTwoFactorAuthState(long principalId, boolean enabled) {
+		if (enabled) {
+			DBOUserTwoFaStatus status = new DBOUserTwoFaStatus();
+			status.setPrincipalId(principalId);
+			status.setEnabled(true);
+			basicDAO.createOrUpdate(status);
```

Comment:
for Mysql it can be faster to update a row.  (Delete and create are more expensive).